src/substrait/textplan/SymbolTablePrinter.cpp

```diff
@@ -141,15 +142,56 @@ std::string typeToText(const ::substrait::proto::Type& type) {
 std::string relationToText(
     const SymbolTable& symbolTable,
     const SymbolInfo& info) {
-  auto relation = ANY_CAST(const ::substrait::proto::Rel*, info.blob);
+  auto relationData = ANY_CAST(std::shared_ptr<RelationData>, info.blob);
 
   PlanPrinterVisitor printer(symbolTable);
-  return printer.printRelation(info.name, relation);
+  return printer.printRelation(info.name, relationData->protoAddr);
+}
+
+std::vector<std::string> followPipeline(
+    const SymbolTable& symbolTable,
+    const ::substrait::proto::Rel* relation) {
+  std::vector<std::string> pipeline;
+  auto info = symbolTable.lookupSymbolByLocation(
+      Location((google::protobuf::Message*)relation));
+  auto relationData = ANY_CAST(std::shared_ptr<RelationData>, info.blob);
+  pipeline.push_back(info.name);
+  if (relationData->continuingPipeline != nullptr) {
+    auto tailPipe =
+        followPipeline(symbolTable, relationData->continuingPipeline);
+    pipeline.insert(pipeline.end(), tailPipe.begin(), tailPipe.end());
+  }
+  return pipeline;
 }
 
 std::string outputPipelinesSection(const SymbolTable& symbolTable) {
-  // TODO: Implement.
-  return "";
+  std::stringstream text;
+  bool hasPreviousText = false;
+  for (const SymbolInfo& info : symbolTable) {
```

Comment:
Somewhat unrelated to printing pipelines but if there is a plan like...

left -> join -> project
right -/

What do the pipelines look like?  Are there two pipelines, one for left and one for right?  Do both include project?

Reply:
Currently that is the form.  The form may need tweaking to make it more readable.